### PR TITLE
Full-body actions and unified sim/real API plus tests

### PR DESCRIPTION
**Pull request by cpaxton**
## Motivation and Context

- Creating single unified API for sim and real actions
- Adding tests for placing and robot motion
- make sure hybrid action space is functional

## Changelog

- add ContinuousFullBodyAction
- update IK code and tests
- adding test code

## How Has This Been Tested

- Real robot experiments
- adding scripts

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

# Checklist:

- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] I ran `tests/hw_manual_test.py` on hardware and observed that robot behavior looks normal.

### CODE REVIEW

File: src/home_robot/home_robot/core/interfaces.py [hunk 1]
```diff
@@ -51,16 +71,27 @@ class HybridAction(Action):
...
+        elif joints is not None:
+            self.action_type = ActionType.CONTINUOUS_MANIPULATION
+            action = ContinuousFullBodyAction(joints, xyt)
+        elif xyt is not None:
             self.action_type = ActionType.CONTINUOUS_NAVIGATION
+            action = ContinuousNavigationAction(xyt)
         else:
-            raise RuntimeError(f"action type{type(action)} not supported")
+            raise RuntimeError("Cannot create HybridAction without any action!")
+        if isinstance(action, HybridAction):
```
Comment (yvsriram) [inline src/home_robot/home_robot/core/interfaces.py:93]: Why can't we just throw an error here?

Comment (yvsriram) [reply src/home_robot/home_robot/core/interfaces.py:93]: When would a HybridAction be passed here?

File: src/home_robot/home_robot/motion/stretch.py [hunk 2]
```diff
@@ -820,6 +822,98 @@ def validate(self, q=None, ignored=[], distance=0.0, verbose=False):
...
+        """Compute deltas"""
+        assert len(deltas) == 10
+        arm = deltas[0] + deltas[1] + deltas[2] + deltas[3]
+        lift = deltas[4]
+        roll = deltas[5]
+        pitch = deltas[6]
+        yaw = deltas[7]
+        pan, tilt = self.head.get_pan_tilt()
+        positions = [
+            0,
```
Comment (yvsriram) [inline src/home_robot/home_robot/motion/stretch.py:886]: What is the 11th joint we are setting to zero everywhere?

**Review by yvsriram — APPROVED** [review]
Looks good to me. Thank you!

File: src/home_robot_hw/home_robot_hw/env/stretch_pick_and_place_env.py [hunk 3]
```diff
@@ -155,15 +190,19 @@ def apply_action(self, action: Action, info: Optional[Dict[str, Any]] = None):
                 # if not self.robot.in_manipulation_mode():
                 #     self.robot.switch_to_manipulation_mode()
                 pass
+            elif action == DiscreteNavigationAction.EXTEND_ARM:
+                """Extend the robot arm"""
```
Comment (zephirefaith) [inline src/home_robot_hw/home_robot_hw/env/stretch_pick_and_place_env.py:194]: Add a warning here for future-proofing.

**Review by zephirefaith — APPROVED** [review]
Looks good. Thanks for adding this. Just to confirm the continuous manipulation action are expected to be given as joint-positions and not EE-goals at the moment, right?

File: src/home_robot_hw/home_robot_hw/env/stretch_pick_and_place_env.py (hunk 3 above)
Comment (cpaxton) [reply src/home_robot_hw/home_robot_hw/env/stretch_pick_and_place_env.py:218]: implemented

**Pull request merged**